jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -159,7 +159,7 @@ def _update_parameters_from_fit(self):
         """Insert the results of the model fit into the component_models"""
         for m in self.component_models:
             name = m["id"]
-            if hasattr(self._fitted_model, name):
+            if hasattr(self._fitted_model, "submodel_names"):
```

Comment:
Looking back at why I made this change for BlackBody, I think you'll need something like the following to handle all cases (current proposed change breaks for adding a BlackBody and Linear component but only fitting the BlackBody component, for example):
```
if name in getattr(self._fitted_model, "submodel_names", []):
```

Reply:
It seems like that case works for me. Are you saying something different than the recording below?

https://user-images.githubusercontent.com/39831871/148075687-042d88e8-abe7-4b50-b54e-33ba7ea29041.mov

Reply:
Ah, it seems that it happens if you have a model with multiple components but also unused components.  See below:

https://user-images.githubusercontent.com/877591/148078170-3eaa3fdb-d075-44f4-acb1-03dc27a96d33.mov

Reply:
Ahh, I see. Thanks for the clarification.